app/Http/Controllers/TechtreeController.php

```diff
@@ -269,9 +269,12 @@ public function getPropertiesTable(GameObject $object, PlayerService $player, Ob
         $properties_array['structural_integrity'] = $structural_integrity->totalValue;
         $properties_array['shield'] = $shield->totalValue;
         $properties_array['attack'] = $attack->totalValue;
-        $properties_array['speed'] = $speed->totalValue;
-        $properties_array['capacity'] = $capacity->totalValue;
-        $properties_array['fuel'] = $fuel->totalValue;
+
+        if ($object->type !== GameObjectType::Defense) {
+            $properties_array['speed'] = $speed->totalValue;
+            $properties_array['capacity'] = $capacity->totalValue;
+            $properties_array['fuel'] = $fuel->totalValue;
+        }
```

Comment:
💡 Could you add this same check to the $tooltips_array lines below?

Because right now tooltips are still being generated for properties that are no longer being shown, resulting in unnecessary calls. Preventing these unnecessary calls should make the code (a tiny bit) more efficient.

Also feel free to refactor the code a bit if you see improvements in readability or otherwise.

Reply:
I've now applied the same check :)